### PR TITLE
Fix OsRealmSchema leak

### PR DESCRIPTION
**Pull request by beeender**

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -411,6 +411,7 @@ private static void initializeSyncedRealm(Realm realm) {
                 realm.setVersion(newVersion);
                 commitChanges = true;
             }
+            schema.close();
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Shouldn't the schema lifecycle follow the Realm instance lifecycle instead? E.g. release it when we release the last instance?

Comment (beeender) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: I have no idea, it seems OsRealmSchema is used for sync schema updating only?

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: If we need `schema.close()` here, we also need to close it at line 405.
Shouldn't we use `try - finally`?

Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/Realm.java]: Yes, right now, but `close()` is on the abstract class, and functionality will probably shift between the two classes, so we cannot make any assumptions

**Pull request merged**